src/random/cpu/choice.cc

```diff
@@ -41,29 +42,29 @@ void RandomEngine::Choice(int64_t num, FloatArray prob, IdxType* out, bool repla
   } else {
     sampler = new utils::TreeSampler<IdxType, FloatType, false>(this, prob);
   }
-  for (int64_t i = 0; i < num; ++i)
+  for (IdxType i = 0; i < num; ++i)
     out[i] = sampler->Draw();
   delete sampler;
 }
 
 template void RandomEngine::Choice<int32_t, float>(
-    int64_t num, FloatArray prob, int32_t* out, bool replace);
+    int32_t num, FloatArray prob, int32_t* out, bool replace);
 template void RandomEngine::Choice<int64_t, float>(
     int64_t num, FloatArray prob, int64_t* out, bool replace);
 template void RandomEngine::Choice<int32_t, double>(
-    int64_t num, FloatArray prob, int32_t* out, bool replace);
+    int32_t num, FloatArray prob, int32_t* out, bool replace);
 template void RandomEngine::Choice<int64_t, double>(
     int64_t num, FloatArray prob, int64_t* out, bool replace);
 
 template <typename IdxType>
-void RandomEngine::UniformChoice(int64_t num, int64_t population, IdxType* out, bool replace) {
+void RandomEngine::UniformChoice(IdxType num, IdxType population, IdxType* out, bool replace) {
   if (!replace)
     CHECK_LE(num, population) << "Cannot take more sample than population when 'replace=false'";
   if (replace) {
-    for (int64_t i = 0; i < num; ++i)
+    for (IdxType i = 0; i < num; ++i)
       out[i] = RandInt(population);
   } else {
-    if (num < population / 2) {
+    if (num < population / 10) {
```

Comment:
Should we leave a TODO here for future tuning?

Reply:
Sure. Will do.